sparkle/src/components/Citation.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { ReactNode } from "react";
+
+import { DocumentText } from "@sparkle/icons/stroke";
+import { classNames } from "@sparkle/lib/utils";
+import { Drive, Github, Notion, Slack } from "@sparkle/logo/platforms";
+
+import { ExternalLinkIcon, Icon, IconButton } from "..";
+
+interface CitationProps {
+  type?: "slack" | "drive" | "github" | "notion" | "document";
+  title: string;
+  description?: string;
+  index?: ReactNode;
+  lastHoveredReference?: number;
+  href: string;
+}
+
+const typeIcons = {
+  slack: Slack,
+  drive: Drive,
+  github: Github,
+  notion: Notion,
+  document: DocumentText,
+};
+
+export function Citation({
+  title,
+  index,
+  type = "document",
+  description,
+  href,
+  lastHoveredReference,
+}: CitationProps) {
+  return (
+    <div
+      className={classNames(
+        "w-48 s-flex s-flex-none s-flex-col s-gap-1 s-rounded-xl s-border s-border-structure-100 s-p-3 s-shadow-sm sm:s-w-64",
+        lastHoveredReference === index ? "animate-[bgblink_500ms_3]" : ""
```

Comment:
nit: isLastHoveredReference, as boolean, marginally better (isolating citation from the citation wrapper)